src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -105,6 +107,9 @@ struct NetworkInfo
 namespace {
 // The internal network info containing credentials. Need to find some better place to save these info.
 NetworkInfo sNetworks[kMaxNetworks];
+
+EmberAfThreadInterfaceScanResult sThreadScanResults[kMaxScannedThreadNetworks];
```

Comment:
`EmberAfThreadInterfaceScanResult` contains just a pointer. Where are we storing the actual data? Perhaps, we should use another data structure here and only construct `EmberAfThreadInterfaceScanResult` when building the response?